homeassistant/components/limitlessled/light.py

```diff
@@ -1,393 +1,107 @@
 """Support for LimitlessLED bulbs."""
-import logging
 
-from limitlessled import Color
-from limitlessled.bridge import Bridge
-from limitlessled.group.dimmer import DimmerGroup
-from limitlessled.group.rgbw import RgbwGroup
-from limitlessled.group.rgbww import RgbwwGroup
-from limitlessled.group.white import WhiteGroup
-from limitlessled.pipeline import Pipeline
-from limitlessled.presets import COLORLOOP
-import voluptuous as vol
+import voluptuous
 
-from homeassistant.components.light import (
-    ATTR_BRIGHTNESS,
-    ATTR_COLOR_TEMP,
-    ATTR_EFFECT,
-    ATTR_FLASH,
-    ATTR_HS_COLOR,
-    ATTR_TRANSITION,
-    EFFECT_COLORLOOP,
-    EFFECT_WHITE,
-    FLASH_LONG,
-    PLATFORM_SCHEMA,
-    SUPPORT_BRIGHTNESS,
-    SUPPORT_COLOR,
-    SUPPORT_COLOR_TEMP,
-    SUPPORT_EFFECT,
-    SUPPORT_FLASH,
-    SUPPORT_TRANSITION,
-    Light,
-)
-from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT, CONF_TYPE, STATE_ON
-import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.restore_state import RestoreEntity
-from homeassistant.util.color import color_hs_to_RGB, color_temperature_mired_to_kelvin
-
-_LOGGER = logging.getLogger(__name__)
-
-CONF_BRIDGES = "bridges"
-CONF_GROUPS = "groups"
-CONF_NUMBER = "number"
-CONF_VERSION = "version"
-CONF_FADE = "fade"
-
-DEFAULT_LED_TYPE = "rgbw"
-DEFAULT_PORT = 5987
-DEFAULT_TRANSITION = 0
-DEFAULT_VERSION = 6
-DEFAULT_FADE = False
-
-LED_TYPE = ["rgbw", "rgbww", "white", "bridge-led", "dimmer"]
+import homeassistant.components.limitlessled.light_rf as light_rf
+import homeassistant.components.limitlessled.light_bridge as light_bridge
 
-EFFECT_NIGHT = "night"
-
-MIN_SATURATION = 10
-
-WHITE = [0, 0]
-
-SUPPORT_LIMITLESSLED_WHITE = (
-    SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP | SUPPORT_EFFECT | SUPPORT_TRANSITION
-)
-SUPPORT_LIMITLESSLED_DIMMER = SUPPORT_BRIGHTNESS | SUPPORT_TRANSITION
-SUPPORT_LIMITLESSLED_RGB = (
-    SUPPORT_BRIGHTNESS
-    | SUPPORT_EFFECT
-    | SUPPORT_FLASH
-    | SUPPORT_COLOR
-    | SUPPORT_TRANSITION
-)
-SUPPORT_LIMITLESSLED_RGBWW = (
-    SUPPORT_BRIGHTNESS
-    | SUPPORT_COLOR_TEMP
-    | SUPPORT_EFFECT
-    | SUPPORT_FLASH
-    | SUPPORT_COLOR
-    | SUPPORT_TRANSITION
-)
+from homeassistant.components.light import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as config_validation
 
+# Validation of the user's configuration
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
Since this integration uses the legacy form of config with a platform config section, we first need to refactor it and move the config to the integration init module and use a `CONFIG_SCHEMA` instead.

See https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md

Please do this first in a separate PR.

Reply:
It seems like this would break the existing configuration files people have written for this integration.

Reply:
Yes, but we don't allow extending platform configuration.